parlai/tasks/huggingface/agents.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import DialogTeacher
+from parlai.utils.data import DatatypeHelper
+from typing import List
+from abc import ABC
+
+# huggingface imports
+from datasets import load_dataset
+
+
+class AbstractHuggingFaceTeacher(DialogTeacher, ABC):
+    """
+    Abstract parent class for HuggingFace teachers. Extend this class and specify the
+    attributes below to use a different dataset.
+
+    hf_path = path parameter passed into hugging face load_dataset function
+    hf_name = name parameter passed into hugging face load_dataset function
+    hf_text_fields = list of names of the data fields from the dataset to be included in the text/query
+    hf_label_field = name of the data field from the hf dataset that specifies the label of the episode
+    hf_splits_mapping = dictionary mapping with the keys 'train', 'valid', and 'test', that map to the
+    names of the splits of the hf dataset.
+    render_text_field = bool where if True, will include the text field name in the query (e.g. "sentence: <sentence>")
+    """
+
+    hf_path: str = 'glue'
+    hf_name: str = 'cola'
+    hf_text_fields: List[str] = ['sentence']
+    hf_label_field: str = 'label'
+    hf_splits_mapping: dict = {'train': 'train', 'valid': 'validation', 'test': 'test'}
+    render_text_field: bool = False
+
+    def __init__(self, opt, shared=None):
+        self.fold = DatatypeHelper.fold(opt['datatype'])
+        self.hf_split = self.hf_splits_mapping[self.fold]
+        opt['datafile'] = self.hf_split
```

Comment:
Make this `os.path.join(opt['datapath'], 'huggingface', self.hf_path, self.hf_name, self.fold)`

Reply:
This file is used for caching the number of items in the dataset and we'll have a name collision otherwise